src/code/z_collision_check.c

```diff
@@ -1378,505 +1559,535 @@ void func_8005E10C(GlobalContext* globalCtx, Collider* collider, Vec3f* pos) {
     D_8015DD68.speed = 8.0f;
     D_8015DD68.gravity = -1.0f;
 
-    Effect_Add(globalCtx, &sp24, EFFECT_SPARK, 0, 1, &D_8015DD68);
+    Effect_Add(globalCtx, &bloodIndex, EFFECT_SPARK, 0, 1, &D_8015DD68);
 }
 
-void func_8005E26C(GlobalContext* globalCtx, Collider* collider, Vec3f* pos) {
+/**
+ * Spawns a burst of water.
+ * Used by collider type HIT4, which no actor has.
+ */
+void CollisionCheck_WaterBurst(GlobalContext* globalCtx, Collider* collider, Vec3f* pos) {
     EffectSsSibuki_SpawnBurst(globalCtx, pos);
-    func_80062B80(globalCtx, pos);
+    CollisionCheck_SpawnWaterDroplets(globalCtx, pos);
 }
 
-void func_8005E2A4(GlobalContext* globalCtx, Collider* collider, Vec3f* v) {
-    func_80062A28(globalCtx, v);
+/**
+ * Spawns red blood drops.
+ * Used by collider type HIT7, which no actor has.
+ */
+void CollisionCheck_RedBlood(GlobalContext* globalCtx, Collider* collider, Vec3f* v) {
+    CollisionCheck_SpawnRedBlood(globalCtx, v);
 }
 
-void func_8005E2C8(GlobalContext* globalCtx, Collider* collider, Vec3f* v) {
-    func_80062A28(globalCtx, v);
+/**
+ * Spawns red blood drops.
+ * Unused.
+ */
+void CollisionCheck_RedBloodUnused(GlobalContext* globalCtx, Collider* collider, Vec3f* v) {
+    CollisionCheck_SpawnRedBlood(globalCtx, v);
 }
 
-void func_8005E2EC(GlobalContext* globalCtx, ColliderBody* colliderBody, Collider* collider, Vec3f* arg3) {
+/**
+ * Plays sound effects and displays hitmarks for solid-type AC colliders (METAL, WOOD, HARD, and TREE)
+ */
+void CollisionCheck_HitSolid(GlobalContext* globalCtx, ColliderInfo* info, Collider* collider, Vec3f* hitPos) {
     s32 flags;
 
-    flags = colliderBody->toucherFlags & 0x18;
-    if (flags == 0 && collider->type != COLTYPE_METAL_SHIELD) {
-        EffectSsHitMark_SpawnFixedScale(globalCtx, 0, arg3);
+    flags = info->toucherFlags & (3 << 3);
```

Comment:
I think `(3 << 3)` doesn't make much sense for this, it's probably supposed to be flags 8 and 0x10 `|`'d together

Reply:
alternatively, it's `TOUCH_SFX_NONE`